From 69dede13028a41f6261fed294e17e8fcf1a789d7 Mon Sep 17 00:00:00 2001
From: Albin B <albin.bala.krishnan@intel.com>
Date: Mon, 31 Aug 2015 22:49:30 +0530
Subject: [PATCH] [EM/PMIC_CCSM]: Clear OTG mode bit during boot

When issue a reboot command with OTG connected and remove the OTG
before platform boots, the OTG mode bit is not getting cleared and
which blocks the charger detection. So clearing OTG mode bit during
boot before checking initial events.

Change-Id: Ib98058ff4d73762a7b5313ee01556a06eeb3b9b0
Tracked-On: https://jira01.devtools.intel.com/browse/GMINL-17488
Signed-off-by: Albin B <albin.bala.krishnan@intel.com>
Signed-off-by: R, Kannappan <r.kannappan@intel.com>
---
 drivers/power/intel_pmic_ccsm.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/power/intel_pmic_ccsm.c b/drivers/power/intel_pmic_ccsm.c
index 8eaedb0c7f54..711171cddd62 100644
--- a/drivers/power/intel_pmic_ccsm.c
+++ b/drivers/power/intel_pmic_ccsm.c
@@ -1902,6 +1902,9 @@ static int pmic_chrgr_probe(struct platform_device *pdev)
 	INIT_DELAYED_WORK(&chc.evt_work, pmic_event_worker);
 	INIT_LIST_HEAD(&chc.evt_queue);
 
+	/* clear otg mode flag before checking events */
+	intel_pmic_handle_otgmode(false);
+
 	ret = pmic_check_initial_events();
 	if (ret)
 		goto otg_req_fail;
-- 
2.6.0.rc0.2.g7662973

